class_main.py

```diff
@@ -0,0 +1,81 @@
+import asyncio
+import json
+import os
+import shutil
+from typing import Tuple
+
+import aiofiles
+
+import GetterPriceBinance
+import mail_handler
+
+
+class main_app:
+    def __new__(cls):
+        """
+        Singleton. Sorry, sometimes I'm stupid.
+        """
+        if not hasattr(cls, "instance"):
+            cls.instance = super(main_app, cls).__new__(cls)
+        return cls.instance
+
+    def __init__(self):
+        self._mail_client = mail_handler.factory_mail_handler(mode="gmail")
+        self._binance_websocket = GetterPriceBinance.book_ticker_price_binance(
+            symbol="BTCUAH"
+        )
+        self._file_with_emails = "subscribed_emails.json"
+        return
+
+    def stop_binance_websocket(self):
+        self._binance_websocket.stop_ws()
+
+    async def get_rate(self) -> float:
+        task_get_price = asyncio.create_task(
+            self._binance_websocket.get_price("BTCUAH")
+        )
+        await task_get_price
+        return task_get_price.result()
+
+    async def _read_file_with_emails(self):
+        data = []
+        if os.path.exists(self._file_with_emails):
```

Comment:
i would better write:
```
if smth not exists:
   return []
...
then continue with a logic
  
```

Reply:
I have smth like this:

`data = []
// some logic if file exist, and this logic can change data
return data`

So, if file exist, data changed. If no, data = [] .

I think, one return is better.